src/modular/div_by_2.rs

```diff
@@ -28,3 +28,23 @@ pub(crate) fn div_by_2<const LIMBS: usize>(a: &Uint<LIMBS>, modulus: &Uint<LIMBS
 
     Uint::<LIMBS>::select(&if_even, &if_odd, is_odd)
 }
+
+#[cfg(feature = "alloc")]
+pub(crate) mod boxed {
```

Comment:
I think putting different variants of this function in modules is a bit of an overcomplication, perhaps just different suffixes would suffice.